framework/src/materials/GenericFunctionVectorMaterial.C

```diff
@@ -0,0 +1,104 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GenericFunctionVectorMaterial.h"
+#include "Function.h"
+
+registerMooseObject("MooseApp", GenericFunctionVectorMaterial);
+registerMooseObject("MooseApp", ADGenericFunctionVectorMaterial);
+
+defineLegacyParams(GenericFunctionVectorMaterial);
+
+template <bool is_ad>
+InputParameters
+GenericFunctionVectorMaterialTempl<is_ad>::validParams()
+{
+
+  InputParameters params = Material::validParams();
+  params.addClassDescription("Material object for declaring vector properties that are populated "
+                             "by evaluation of Function objects.");
+  params.addParam<std::vector<std::string>>("prop_names",
+                                            "The names of the properties this material will have");
+  params.addParam<std::vector<FunctionName>>("prop_values",
+                                             "The corresponding names of the "
+                                             "functions that are going to provide "
+                                             "the values for the variables, "
+                                             "minor ordering by component");
+  params.addDeprecatedParam<bool>("enable_stateful",
+                                  false,
+                                  "Enable the declaration of old and older values",
+                                  "all properties can implicitly become stateful");
+  return params;
+}
+
+template <bool is_ad>
+GenericFunctionVectorMaterialTempl<is_ad>::GenericFunctionVectorMaterialTempl(
+    const InputParameters & parameters)
+  : Material(parameters),
+    _prop_names(getParam<std::vector<std::string>>("prop_names")),
+    _prop_values(getParam<std::vector<FunctionName>>("prop_values")),
+    _enable_stateful(getParam<bool>("enable_stateful"))
+{
+  unsigned int num_names = _prop_names.size();
+  unsigned int num_values = _prop_values.size();
+
+  if (num_names * LIBMESH_DIM != num_values)
+    mooseError("Number of prop_names (",
+               num_names,
+               ") times the libmesh dimension (",
+               LIBMESH_DIM,
+               ") must match the number of prop_values (",
+               num_values,
+               ") for a GenericFunctionVectorMaterial!");
+
+  _num_props = num_names;
+
+  _properties.resize(num_names);
+
+  if (_enable_stateful)
+  {
+    _properties_old.resize(num_names);
+    _properties_older.resize(num_names);
+  }
```

Comment:
this is the same code as in GenericFunctionMaterial and it s all there is

I wonder if we re missing some code there?

Reply:
it s deleted. no failures. So that was deprecated code.